sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -55,6 +69,22 @@ public HttpMessage CreateMessage()
             return new HttpMessage(CreateRequest(), ResponseClassifier);
         }
 
+        /// <summary>
+        /// Creates a new <see cref="HttpMessage"/> instance.
+        /// </summary>
+        /// <param name="context">Context specifying the message options.</param>
+        /// <returns>The message.</returns>
+        public HttpMessage CreateMessage(RequestContext context)
+        {
+            var message = CreateMessage();
+            if (context != null)
+            {
+                message.CancellationToken = context.CancellationToken;
+                message.Policies = context.Policies;
```

Comment:
Is it ok that we provide direct reference to a list that can be modified in parallel with `SendAsync`?

Reply:
I'll add the policies instead.  Thanks!